### PR TITLE
OTE_SDK entities. Increase code coverage.

### PR DESCRIPTION
**Pull request by dvkruchinin**
Test list:
ote_sdk/ote_sdk/tests/entities/test_color.py
ote_sdk/ote_sdk/tests/entities/test_label.py
ote_sdk/ote_sdk/tests/entities/test_resultset.py
ote_sdk/ote_sdk/tests/entities/test_scored_label.py
ote_sdk/ote_sdk/tests/entities/test_task_environment.py

CVS-71933

### CODE REVIEW

**Comment by dvkruchinin** [comment]
run ote_sdk tests

File: ote_sdk/ote_sdk/tests/entities/test_resultset.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        assert type(result_set.creation_date) == datetime
+        assert result_set.id == ID()
+
+        assert result_set.has_score_metric() is False
+        result_set.performance = "test_performance"
+        assert result_set.performance != NullPerformance()
+        assert result_set.has_score_metric() is True
+
+        result_set.creation_date = now().replace(microsecond=0)
+        assert result_set.creation_date == now().replace(microsecond=0)
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_resultset.py:109]: I guess the issue still exists

Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_resultset.py]: The error was on line 99, but even here it was replaced by a specific time.

**Comment by dvkruchinin** [comment]
run ote_sdk tests

File: ote_sdk/ote_sdk/tests/entities/test_resultset.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            "prediction_dataset": None,
+            "purpose": None,
+            "performance": None,
+            "creation_date": None,
+            "id": None,
+        }
+
+        result_set = ResultSetEntity(**test_data)
+
+        for name, value in test_data.items():
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_resultset.py:86]: You compare attributes of result_set with `None` always. Could you make "model" to be instance of ModelEntity, "ground_truth_dataset" to be an instance of DatasetEntity and so on?

File: ote_sdk/ote_sdk/tests/entities/test_resultset.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        assert result_set.performance == NullPerformance()
+        assert type(result_set.creation_date) == datetime
+        assert result_set.id == ID()
+
+        assert result_set.has_score_metric() is False
+        result_set.performance = "test_performance"
+        assert result_set.performance != NullPerformance()
+        assert result_set.has_score_metric() is True
+
+        result_set.creation_date = "2021-11-30 09:07:31+00:00"
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_resultset.py:107]: please use proper type

File: ote_sdk/ote_sdk/tests/entities/test_resultset.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        assert result_set.creation_date == "2021-11-30 09:07:31+00:00"
+
+        set_attr_id = "123456789"
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_resultset.py:110]: ID(...)

File: ote_sdk/ote_sdk/tests/entities/test_scored_label.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        Check the ScoredLabel can correctly return the value
+
+        <b>Input data:</b>
+        LabelEntity
+
+        <b>Expected results:</b>
+        Test passes if the results match
+        """
+        car = LabelEntity(
+            id=123456789, name="car", domain=Domain.DETECTION, is_empty=True
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_scored_label.py:41]: ID(...)

File: ote_sdk/ote_sdk/tests/entities/test_scored_label.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        )
+        person = LabelEntity(
+            id=987654321, name="person", domain=Domain.DETECTION, is_empty=True
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_scored_label.py:44]: ID(...)

File: ote_sdk/ote_sdk/tests/entities/test_task_environment.py [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+    cur_config = __get_path_to_file("./dummy_config.yaml")
+    with open(cur_config, "r") as stream:
+        return yaml.safe_load(stream)
+
+
+def environment():
+    """
+    Return TaskEnvironment
+    """
+    labels_list = []
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_task_environment.py:50]: Can it be non-empty?

File: ote_sdk/ote_sdk/tests/entities/test_task_environment.py [hunk 8]
```diff
@@ -0,0 +1,450 @@
...
+
+        <b>Steps</b>
+        1. Checking parameters after setting
+        """
+        env = environment()
+
+        header = "Test header"
+        description = "Test description"
+        visible_in_ui = False
+        id = "123456789"
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_task_environment.py:426]: ID(...)

**Review by Ilya-Krylov — CHANGES_REQUESTED** [review]

File: ote_sdk/ote_sdk/tests/entities/test_resultset.py (hunk 2 above)
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_resultset.py:86]: Here, the possibility of assigning an attribute is simply checked. Is it worth complicating the test so much?

**Comment by dvkruchinin** [comment]
run ote_sdk tests

File: ote_sdk/ote_sdk/tests/entities/test_label.py [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+        "name": "person",
+        "domain": Domain.DETECTION,
+        "color": "#ff1111",
+        "hotkey": "ctrl+2",
+        "creation_date": datetime.datetime.today(),
+        "is_empty": False,
+        "id": 987654321,
+    }
+
+    car = LabelEntity(**label_car_params)  # type: ignore
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_label.py]: The comment was deleted.

File: ote_sdk/ote_sdk/tests/entities/test_label.py [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+        "name": "car",
+        "domain": Domain.DETECTION,
+        "color": "#ff0000",
+        "hotkey": "ctrl+1",
+        "creation_date": datetime.datetime.today(),
+        "is_empty": False,
+        "id": 123456789,
+    }
+
+    other_label_car_params = {
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_label.py]: Sure.

File: ote_sdk/ote_sdk/tests/entities/test_label.py [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+            "hotkey",
+            "creation_date",
+            "is_empty",
+            "id",
+        ]:
+            assert getattr(self.car, attr) == self.label_car_params[attr]
+
+        label_car_new_name = "electric car"
+        label_car_new_domain = Domain.CLASSIFICATION
+        label_car_new_color = "#00ff00"
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_label.py]: Done

File: ote_sdk/ote_sdk/tests/entities/test_label.py [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+        label_car_new_hotkey = "ctrl+2"
+        label_car_new_id = 987654321
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_label.py]: Done

File: ote_sdk/ote_sdk/tests/entities/test_label.py [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+        assert len(domain) == 6
+
+
+@pytest.mark.components(OteSdkComponent.OTE_SDK)
+class TestLabelEntity:
+
+    label_car_params = {
+        "name": "car",
+        "domain": Domain.DETECTION,
+        "color": "#ff0000",
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_label.py]: Done

File: ote_sdk/ote_sdk/tests/entities/test_label.py [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+        "hotkey": "ctrl+1",
+        "creation_date": datetime.datetime.today(),
+        "is_empty": False,
+        "id": 123456789,
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_label.py]: Done

File: ote_sdk/ote_sdk/tests/entities/test_label.py [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+    }
+
+    other_label_car_params = {
+        "name": "person",
+        "domain": Domain.DETECTION,
+        "color": "#ff1111",
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_label.py]: Done

File: ote_sdk/ote_sdk/tests/entities/test_label.py [hunk 16]
```diff
@@ -0,0 +1,182 @@
...
+        "hotkey": "ctrl+2",
+        "creation_date": datetime.datetime.today(),
+        "is_empty": False,
+        "id": 987654321,
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_label.py]: Done

File: ote_sdk/ote_sdk/tests/entities/test_label.py [hunk 17]
```diff
@@ -0,0 +1,182 @@
...
+
+        <b>Expected results:</b>
+        Test passes if incoming data is processed correctly
+
+        <b>Steps</b>
+        1. Use already created dummy data
+        2. Check the processing of changed id
+        """
+
+        self.empty.id = 999999999
```
[the remark this answers is not in the document]
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_label.py]: Done

File: ote_sdk/ote_sdk/tests/entities/test_resultset.py (hunk 3 above)
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_resultset.py]: Changed to the current date without taking into account seconds and milliseconds

File: ote_sdk/ote_sdk/tests/entities/test_resultset.py (hunk 4 above)
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_resultset.py]: Done

File: ote_sdk/ote_sdk/tests/entities/test_scored_label.py (hunk 5 above)
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_scored_label.py]: Done

File: ote_sdk/ote_sdk/tests/entities/test_scored_label.py (hunk 6 above)
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_scored_label.py]: Done

File: ote_sdk/ote_sdk/tests/entities/test_task_environment.py (hunk 7 above)
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_task_environment.py]: Sure. Reworked.

File: ote_sdk/ote_sdk/tests/entities/test_task_environment.py (hunk 8 above)
Comment (dvkruchinin) [reply ote_sdk/ote_sdk/tests/entities/test_task_environment.py]: Done

**Comment by dvkruchinin** [comment]
run ote_sdk tests

**Comment by dvkruchinin** [comment]
run ote_sdk tests

**Review by Ilya-Krylov — APPROVED** [review]

**Pull request merged**